repos/system_upgrade/common/actors/initramfs/checkfipsenabled/actor.py

```diff
@@ -0,0 +1,33 @@
+from leapp.actors import Actor
+from leapp.exceptions import StopActorExecutionError
+from leapp.models import FIPSInfo
+from leapp.tags import IPUWorkflowTag, LateTestsPhaseTag
+
+
+class CheckFIPSCorrectlyEnabled(Actor):
+    """
+    Sanity check to stop the IPU if the system did not boot into the upgrade initramfs with FIPS settings preserved.
+
+    The performed check should be unlikely to fail, as it would mean that the upgrade boot entry was created without
+    fips=1 on the kernel cmdline.
+    """
+
+    name = 'check_fips_correctly_enabled'
+    consumes = (FIPSInfo,)
+    produces = ()
+    tags = (LateTestsPhaseTag, IPUWorkflowTag)
+
+    def process(self):
+        fips_info = next(self.consume(FIPSInfo), None)
+        if not fips_info:
+            # Unexpected, FIPSInfo is produced unconditionally
+            raise StopActorExecutionError('Cannot check for the correct FIPS state in the upgrade initramfs',
+                                          details={'Problem': 'Did not receive any FIPSInfo message'})
+
+        if fips_info.is_enabled:
+            with open('/proc/sys/crypto/fips_enabled') as fips_status_handle:
+                fips_status = fips_status_handle.read().strip()
+                if fips_status == '1':
```

Comment:
We want `/proc/sys/crypto/fips_enabled` to contain 1, that tells us that FIPS mode is enabled.

```suggestion
                if fips_status != '1':
```

Reply:
Thank you. Fixed with latest changes.